test/integration/src/main/java/io/pravega/test/integration/selftest/adapters/BookKeeperAdapter.java

```diff
@@ -0,0 +1,271 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.pravega.test.integration.selftest.adapters;
+
+import com.google.common.base.Preconditions;
+import io.pravega.common.Exceptions;
+import io.pravega.common.concurrent.ExecutorServiceHelpers;
+import io.pravega.common.concurrent.FutureHelpers;
+import io.pravega.common.lang.ProcessStarter;
+import io.pravega.common.util.ArrayView;
+import io.pravega.segmentstore.contracts.StreamSegmentExistsException;
+import io.pravega.segmentstore.contracts.StreamSegmentNotExistsException;
+import io.pravega.segmentstore.storage.DurableDataLog;
+import io.pravega.segmentstore.storage.DurableDataLogException;
+import io.pravega.segmentstore.storage.impl.bookkeeper.BookKeeperConfig;
+import io.pravega.segmentstore.storage.impl.bookkeeper.BookKeeperLogFactory;
+import io.pravega.segmentstore.storage.impl.bookkeeper.BookKeeperServiceRunner;
+import io.pravega.segmentstore.storage.impl.bookkeeper.ZooKeeperServiceRunner;
+import io.pravega.test.integration.selftest.Event;
+import io.pravega.test.integration.selftest.TestConfig;
+import io.pravega.test.integration.selftest.TestLogger;
+import java.io.File;
+import java.io.IOException;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.atomic.AtomicBoolean;
+import javax.annotation.concurrent.GuardedBy;
+import lombok.val;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.CuratorFrameworkFactory;
+import org.apache.curator.retry.ExponentialBackoffRetry;
+
+/**
+ * Store adapter that executes requests directly to BookKeeper via the BookKeeperLog class.
+ */
+class BookKeeperAdapter implements StoreAdapter {
+    //region Members
+
+    private static final String LOG_ID = BookKeeperAdapter.class.getSimpleName();
+    private final TestConfig testConfig;
+    private final BookKeeperConfig bkConfig;
+    private final ScheduledExecutorService executor;
+    private final ConcurrentHashMap<String, DurableDataLog> logs;
+    @GuardedBy("internalIds")
+    private final HashMap<String, Integer> internalIds;
+    private final AtomicBoolean closed;
+    private final Thread stopBookKeeperProcess;
+    private Process bookKeeperService;
```

Comment:
THese three either need to be guarded by a lock or this class should be made into a service, which will handle this for you (assuming they are initialized in doSstart)